### DISCUSSION TITLE
pnpm up --interactive does not take into account overrides, and I have to go through overrides and update packages manually

### DISCUSSION THREAD

**Discussion by JustFly1984** [pnpm/pnpm#16116] [Ideas]
### Verify latest release

- [X] I verified that the issue exists in the latest pnpm release

### pnpm version

9.12.2

### Which area(s) of pnpm are affected? (leave empty if unsure)

Dependencies resolver, CLI, Lockfile

### Reproduction steps

For example I have minimal packaje.json
```
{
  dependencies: {
     "@types/react": "18.3.11",
  },
  "pnpm: {
      "overrides": {
           "@types/react": "18.3.11",
      }
  }
}
```

I run `pnpm up --interactive`, and it finds correctly that @types/react latest version is 18.3.2 and offers to update, but updated only dependencies, doesn't take in account overrides.

### Describe the Bug

In most cases I've used overrides to enforce latest version. In very rare cases I have to downgrade some package, cos it bugged in latest version.

Yarn upgrade interactive shows overrides packages which is possible to upgrade, but pnpm does not.

### Expected Behavior

expected behavior - take overrides packages into account while `pnpm up --interactive"

### Which Node.js version are you using?

22.10.0

### Which operating systems have you used?

- [X] macOS
- [ ] Windows
- [ ] Linux

**Comment by hedefalk** [comment]
+1

On a workspace monorepo I have a lot of overrides in base package.json so all subpackages use the same versions. `pnpm up` doesn't wanna touch those :(

**Comment by SynthLuvr** [comment]
I'm working on a fix for this.

Current behavior (reproduced on v11.28 and v12.7): when a direct dependency is governed by an override, `pnpm up --latest <pkg>` (and selecting the package in `pnpm up --interactive`) never touches the override, so the update cannot take effect. v11 silently no-ops; v12 additionally rewrites the `package.json` range to the latest version while resolution stays pinned by the override, leaving the manifest and lockfile inconsistent.

Planned behavior: an update that explicitly targets a dependency by name moves the override that pins it (keeping the override's own range style: an exact `1.0.0` pin becomes the new exact version, `^1.0.0` becomes the new `^` range), alongside the lockfile. Updates that cannot move the override (e.g. a `catalog:`, `link:`, or `$ref` value, or a `pkg@range`-scoped selector) will warn instead of failing silently.

---
Written by an agent (goose, claude-opus-4-8).
